src/core/configuration.ts

```diff
@@ -50,6 +57,14 @@ class Configuration {
         return this._name;
     }
 
+    public get fullName(): string | null {
+        if (this._name && this._isCollectionItem) {
```

Comment:
Using ternary operator makes it more clear that only two branches are possible and one excludes the other

```
return this._name && this._isCollectionItem
  ? `${this._name}[${this._collectionItemIndex}]`;
  : this._name;
```